### ISSUE TITLE
Is Text2Mel based on deep convolutional neural networks (CNN) compatible with ParallelWaveGAN? 

### ISSUE THREAD

**Issue by ican24** [kan-bayashi/ParallelWaveGAN#139]
Dear @kan-bayashi,

Thank you for your great project!

Is there a chance to use Text2Mel based on deep convolutional neural networks (CNN) with your ParallelWaveGAN?
It is a standard pythorch .pth file which was trained on the platform like https://github.com/tugstugi/pytorch-dc-tts.
My destination language is so specific to reach good results with Tacotron2. Therefore I am mainly using pytorch-dc-tts for my TTS development.
 
Thank you in advance

**Comment by kan-bayashi** [comment]
Of course, you can do it.
You should check the following points to use your CNN-based Text2Mel model with PWG:
- Feature setting
	- FFT size / hop size / window size
	- The Mel basis (I use librosa-based Mel basis)
	- The range of Mel (I use 80-7600 Hz as default)
- Normalization
	- I use mean-var normalization (i.e., StandardScaler)

If the feature extraction part is different, you need to unify it.
But if you use the same Mel-spectrogram & normalization method for both Text2Mel and PWG, it should be working.

**Comment by ican24** [comment]
Thank you very much!
I am going to try it

**Comment by ican24** [comment]
Dear @kan-bayashi,

May I ask any plain template of Text2Mel + ParallelWaveGAN: a rough schematic example to synthesize?   
Unfortunately I could not combine them learning your codes from tts_realtime_demo.ipynb and https://github.com/erogol/ParallelWaveGAN because of low Deep Learning skills (my experience in Deep Learning is just 2-3 months).  
I believe that such solution may popularize your ParallelWaveGAN solution for many developers like me and stimulate wide usage in.

Thank you in advance!
You are really doing a great job!

**Comment by kan-bayashi** [comment]
I could not understand what you want.
The notebook is the simplest for me, so if you cannot understand it, I have no idea.

**Comment by ican24** [comment]
Yes, you notebook is simplest, but it loads Tacotron2 model.
I can not replace it with  Text2Mel model. There are lot of architecture differences between  Tacotron2 and simple Text2Mel.
I just offered to design or advise how we can replace Tacotron2 model with Text2Mel in you notebook example to save many-many hours of experiments, if it is possible.

**Comment by kan-bayashi** [comment]
I don’t know what is “simple Text2Mel” but I think its input and output is almost the same as Tacotron 2 (i.e. text id sequence and mel-spectrogram). So what you need is just replacing the model 
, right? Or you want to know how to design the simple Text2Mel model architecture?

**Comment by ican24** [comment]
Ok, give me, please, 5 minutes to learn my task.
Here is a code, which is so close to mine.
https://github.com/tugstugi/pytorch-dc-tts/blob/master/synthesize.py
I would like to replace ssrn model by ParallelWaveGAN in this framework or your.
Is it possible?
Thank you in advance

**Comment by kan-bayashi** [comment]
For me, it is very clear. Z seems mel-spectrogram, which is the inputs of PWG.
Anyway, your question is out of support of this repository.
I am not free customer support.
Please ask your colleagues about the further questions.

**Comment by tsehaysun** [comment]
> 

Helllo ,
I am facing the same issue as you mentioned. have you been able to use Text2Mel plus wavegan ?

How can I use wavegan to synthesize texts ?

Best Regards
Tsehayneh M
